logbus/logstreamer/logstreamer.go

```diff
@@ -17,29 +20,149 @@ import (
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+const (
+	DefaultChSize = 10240
+	drainTimeout  = 60 * time.Second
+)
+
+type CloudClient interface {
+	StreamLogs(ctx context.Context, buildID string, deltas cloud.Deltas) error
+}
+
+func decongest(ch <-chan []*logstream.Delta) {
```

Comment:
nit:
Since `decongest` and `deltasIter` are internal implementation details - does it make sense to move this code below the logstreamer?

Reply:
That's fair. I like to organize my code such that a person could feasibly read from top to bottom - I find it a little jarring to jump back and forth between definitions too often. So I put these at the top so that a reader would have already understood what they do by the time they get to the place where they are used (in the LogStreamer).

However, that's just my own personal style, and I'm not particularly married to it. Keeping exported code above unexported code is also a valid style, and I'd rather be consistent with the project than stick to my own personal choices.